pkg/kubelet/kubelet_pods.go

```diff
@@ -1624,6 +1624,26 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 		if err != nil {
 			klog.V(4).InfoS("Cannot get host IPs", "err", err)
 		} else {
+			if s.HostIP != "" {
+				ok, err := utilnet.IsDualStackIPStrings([]string{s.HostIP, hostIPs[0].String()})
+				if err != nil {
+					kl.recorder.Eventf(pod, v1.EventTypeWarning, "InvalidHostIPs", "host IPs are invalid %q or %q", s.HostIP, hostIPs[0].String())
+				}
+				if ok {
```

Comment:
I think

    if utilnet.IPFamilyOfString(s.HostIP) != utilnet.IPFamilyOf(hostIPs[0]) {

is clearer: the error case can't actually happen anyway, and we're comparing exactly two IPs, not scanning a possibly-long list of IPs.

Reply:
I also prefer clearer code. Updated.